modin/tests/pandas/utils.py

```diff
@@ -1097,14 +1199,24 @@ def create_test_dfs(
         post_fn = lambda df: post_fn(df).convert_dtypes(  # noqa: E731
             dtype_backend=backend
         )
-    return tuple(
+    if data_frame_mode:
+        actual_data_frame_mode = NativeDataframeMode().get()
+        NativeDataframeMode().put(data_frame_mode)
+    test_dfs = tuple(
         map(post_fn, [pd.DataFrame(*args, **kwargs), pandas.DataFrame(*args, **kwargs)])
     )
+    if data_frame_mode:
+        NativeDataframeMode().put(actual_data_frame_mode)
+
+    return test_dfs
 
 
 def create_test_series(
-    vals, sort=False, backend=None, **kwargs
+    vals, sort=False, backend=None, data_frame_mode=None, **kwargs
 ) -> tuple[pd.Series, pandas.Series]:
+    if data_frame_mode:
+        actual_data_frame_mode = NativeDataframeMode().get()
+        NativeDataframeMode().put(data_frame_mode)
```

Comment:
@arunjose696 I would recommend doing this at a higher level, using fixture `modify_config`.

Reply:
It would not be possible to use this in the fixture , as the purpose of this is to set two different NativeDataframeMode for 2 different dataframes in the pytest, We dont want to be setting the NativeDataframeMode in the fixture as these dataframes wont be created in the fixture.

Reply:
In that case let's create `create_test_series_in_defined_mode` function in `modin/tests/pandas/native_df_mode/utils.py`. Something like:
```python
def create_test_series_in_defined_mode(vals, sort=False, backend=None, df_mode=None, **kwargs):
    with context(NativeDataframeMode=df_mode):
        return create_test_series(vals, sort=False, backend=None, **kwargs)
```

And similar for the dataframes. 

The main motivation is not to introduce functionality where it is rarely used. Also, given that there is a separate folder with tests, it seems better to try to localize the necessary changes.